### PR TITLE
Crowdsec add an Unban Page

### PR DESCRIPTION
**Pull request by stephdl**
see https://trello.com/c/l85To6SL/358-crowdsec-p2-ui-display-active-bans

The purpose is to make a list of active current banned ip and allow the sysadmin to review them, then he could release all IP or just release one

linked to https://github.com/NethServer/ns8-crowdsec/pull/21

image
image
image
image
image
image
image

We retrieve informations from 

```
[root@R2 ~]# cscli decisions list --output json
[
 {
  "capacity": 0,
  "created_at": "2023-08-17T13:15:42Z",
  "decisions": [
   {
    "duration": "3h56m17.774881231s",
    "id": 99032,
    "origin": "cscli",
    "scenario": "manual 'ban' from 'localhost'",
    "scope": "Ip",
    "simulated": false,
    "type": "ban",
    "value": "1.2.3.54"
   }
  ],
  "events": null,
  "events_count": 1,
  "id": 41,
  "labels": null,
  "leakspeed": "0",
  "machine_id": "localhost",
  "message": "manual 'ban' from 'localhost'",
  "scenario": "manual 'ban' from 'localhost'",
  "scenario_hash": "",
  "scenario_version": "",
  "simulated": false,
  "source": {
   "ip": "1.2.3.54",
   "scope": "Ip",
   "value": "1.2.3.54"
  },
  "start_at": "2023-08-17 13:15:42 +0000 UTC",
  "stop_at": "2023-08-17 13:15:42 +0000 UTC",
  "uuid": "6d0ddef4-bf02-4576-92f3-6a4f64a99fd2"
 }
]
```

### CODE REVIEW

File: ui/public/i18n/en/translation.json [hunk 1]
```diff
@@ -56,6 +56,24 @@
     "smarthost_is_disabled": "Email notifications are disabled",
     "smarthosts_is_needed_to_send_notifications": "Enable Smarthost to receive CrowdSec email notifications"
   },
+  "unban":{
+    "title": "Unban IP",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:60]:
```suggestion
    "title": "Banned IP addresses",
```

File: ui/public/i18n/en/translation.json [hunk 2]
```diff
@@ -56,6 +56,24 @@
...
+    "unban_tooltip": "Crowdsec bans IP that have made some bad login attempts to your server, you can release them here",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:61]:
```suggestion
    "unban_tooltip": "IP addresses that failed multiple login attempts are listed here. You can unban them if needed",
```

File: ui/public/i18n/en/translation.json [hunk 3]
```diff
@@ -56,6 +56,24 @@
...
+    "no_bans": "You have no bans, this is a normal behavior",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:62]:
```suggestion
    "no_bans": "No active bans",
```

File: ui/public/i18n/en/translation.json [hunk 4]
```diff
@@ -56,6 +56,24 @@
...
+    "reload_bans": "Reload the list",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:63]:
```suggestion
    "reload_bans": "Refresh",
```

File: ui/public/i18n/en/translation.json [hunk 5]
```diff
@@ -56,6 +56,24 @@
...
+    "delete_all": "Release all IP",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:64]:
```suggestion
    "delete_all": "Unban all",
```

File: ui/public/i18n/en/translation.json [hunk 6]
```diff
@@ -56,6 +56,24 @@
...
+    "delete": "Release",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:65]:
```suggestion
    "delete": "Unban",
```

File: ui/public/i18n/en/translation.json [hunk 7]
```diff
@@ -56,6 +56,24 @@
...
+    "search_bans": "Search",
+    "col_created_at": "Created",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:67]:
```suggestion
    "col_created_at": "Banned",
```

File: ui/public/i18n/en/translation.json [hunk 8]
```diff
@@ -56,6 +56,24 @@
...
+    "col_value": "IP",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:68]:
```suggestion
    "col_value": "IP address",
```

File: ui/public/i18n/en/translation.json [hunk 9]
```diff
@@ -56,6 +56,24 @@
...
+    "col_duration": "Ban time",
+    "col_scenario": "Reason",
+    "unban_ip": "Unban IP",
+    "unban_ip_confirm": "Confirm the release of the banned IP: <strong>{name}</strong>?",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:72]:
```suggestion
    "unban_ip_confirm": "Unban IP address <strong>{name}</strong>?",
```

File: ui/public/i18n/en/translation.json [hunk 10]
```diff
@@ -56,6 +56,24 @@
...
+    "confirm_unban_ip_message": "The IP will be released are you sure ?",
+    "confirm_delete_all_bans_message": "Confirm the release of the all banned IP",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:74]:
```suggestion
    "confirm_delete_all_bans_message": "Unban all IP addresses?",
```

File: ui/public/i18n/en/translation.json [hunk 11]
```diff
@@ -56,6 +56,24 @@
...
+    "delete_bans": "Release all banned IP"
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:75]:
```suggestion
    "delete_bans": "Unban all"
```

File: ui/public/i18n/en/translation.json [hunk 12]
```diff
@@ -70,7 +88,9 @@
     "get-name": "Get name",
     "list-backup-repositories": "List backup repositories",
     "list-backups": "List backups",
-    "list-installed-modules": "List installed modules"
+    "list-installed-modules": "List installed modules",
+    "unban_ip_status": "List the banned IP",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:92]:
```suggestion
    "unban_ip_status": "List banned IP addresses",
```

File: ui/public/i18n/en/translation.json [hunk 13]
```diff
@@ -70,7 +88,9 @@
...
+    "unban-ip": "Release one banned IP"
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:93]:
```suggestion
    "unban-ip": "Unban IP address"
```

**Review by andre8244 — CHANGES_REQUESTED** [review]
- I would not use a "danger" button for "Release/unban all IP addresses", since it's not a destructive action
- I would not use a "danger" modal to unban one or all IP aderesses, but a standard `NsModal` with message "Unban IP address xyz?" and two buttons "Cancel" and "Unban"

File: ui/public/i18n/en/translation.json [hunk 14]
```diff
@@ -56,6 +56,24 @@
...
+    "reload_bans": "Refresh",
+    "delete_all": "Unban all",
+    "delete": "Unban",
+    "search_bans": "Search",
+    "col_created_at": "Banned",
+    "col_value": "IP address",
+    "col_duration": "Ban time",
+    "col_scenario": "Reason",
+    "unban_ip": "Unban IP",
+    "confirm_unban_ip_message": "The IP will be released are you sure ?",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:72]: Let's always use the same glossary

```suggestion
    "confirm_unban_ip_message": "Unban IP address?",
```

File: ui/public/i18n/en/translation.json [hunk 15]
```diff
@@ -56,6 +56,24 @@
...
+    "confirm_delete_all_bans_message": "Unban all IP addresses?",
+    "release_all_ip": "Release all banned IP",
```
Comment (andre8244) [inline ui/public/i18n/en/translation.json:74]:
```suggestion
    "release_all_ip": "Unban all IP addresses",
```

File: ui/src/components/ConfirmReleaseIP.vue [hunk 16]
```diff
@@ -0,0 +1,38 @@
...
+    @primary-click="$emit('confirm')"
+  >
+    <template slot="title"
+      >{{ $t("unban.unban_ip") }}: {{ ban.value }}</template
+    >
+    <template slot="content">
+      <div>{{ $t("unban.confirm_unban_ip_message") }}</div>
+    </template>
+    <template slot="secondary-button">{{ core.$t("common.cancel") }}</template>
+    <template slot="primary-button">{{ core.$t("common.delete") }}</template>
```
Comment (andre8244) [inline ui/src/components/ConfirmReleaseIP.vue:19]:
```suggestion
    <template slot="primary-button">{{ core.$t("unban.delete") }}</template>
```

File: ui/src/components/ConfirmReleaseIPAll.vue [hunk 17]
```diff
@@ -0,0 +1,35 @@
...
+    :visible="isShown"
+    @modal-hidden="$emit('hide')"
+    @primary-click="$emit('confirm')"
+  >
+    <template slot="title">{{ $t("unban.release_all_ip") }}</template>
+    <template slot="content">
+      <div>{{ $t("unban.confirm_delete_all_bans_message") }}</div>
+    </template>
+    <template slot="secondary-button">{{ core.$t("common.cancel") }}</template>
+    <template slot="primary-button">{{ core.$t("common.delete") }}</template>
```
Comment (andre8244) [inline ui/src/components/ConfirmReleaseIPAll.vue:17]:
```suggestion
    <template slot="primary-button">{{ core.$t("unban.delete") }}</template>
```

File: ui/src/components/ConfirmReleaseIP.vue (hunk 16 above)
Comment (stephdl) [reply ui/src/components/ConfirmReleaseIP.vue:19]: are you sure for this translation ?

Comment (andre8244) [reply ui/src/components/ConfirmReleaseIP.vue:19]: Why not? The user just clicked "Unban" on the kebab menu that shows this modal, it is a good a practice to repeat the same verb on the confirmation button

**Pull request merged**